### PR TITLE
.Net: Remove SKParameterMetadata.Type and ParameterJsonType

### PR DESCRIPTION
**Pull request by stephentoub**
It's now duplicative of information in the schema.
Fixes https://github.com/microsoft/semantic-kernel/issues/3544

### CODE REVIEW

File: dotnet/src/Planners/Planners.Core/Handlebars/HandlebarsTemplateEngineExtensions.cs [hunk 1]
```diff
@@ -378,14 +358,20 @@ private static double CastToNumber(object? number)
      * Handlebar argument type is an object (this covers complex types).
      * Function parameter is a generic type.
      */
-    private static bool IsExpectedParameterType(string functionViewType, string handlebarArgumentType, object handlebarArgValue)
+    private static bool IsExpectedParameterType(SKParameterMetadata parameterType, object argument)
```
Comment (stephentoub) [inline dotnet/src/Planners/Planners.Core/Handlebars/HandlebarsTemplateEngineExtensions.cs:361]: I made a best guess on this code, but I'm not sure if/how the previous code was actually correct, and it doesn't look like tests are exercising any of this.

File: dotnet/src/Connectors/Connectors.AI.OpenAI/AzureSdk/SKFunctionMetadataExtensions.cs [hunk 2]
```diff
@@ -12,34 +13,51 @@ public static class SKFunctionMetadataExtensions
...
+            SKJsonSchema? schema = param.Schema ?? OpenAIFunction.GetJsonSchema(param.ParameterType, param.Description);
+
+            // Read the "type" property from the schema, if it exists.
+            // If it doesn't, that means we lack any type information for the property.
+            string? type = null;
+            if (schema?.RootElement.TryGetProperty("type", out JsonElement prop) == true)
+            {
+                type = prop.GetString();
+            }
+            type ??= "string";
```
Comment (alliscode) [inline dotnet/src/Connectors/Connectors.AI.OpenAI/AzureSdk/SKFunctionMetadataExtensions.cs:34]: Is `type` used for anything?

**Review by alliscode — APPROVED** [review]

File: dotnet/src/Connectors/Connectors.AI.OpenAI/AzureSdk/SKFunctionMetadataExtensions.cs (hunk 2 above)
Comment (stephentoub) [reply dotnet/src/Connectors/Connectors.AI.OpenAI/AzureSdk/SKFunctionMetadataExtensions.cs:34]: Hah, it was, and then I subsequently deleted OpenAIFunctionParameter.Type, at which point this is dead.

File: dotnet/src/Planners/Planners.Core/Handlebars/HandlebarsTemplateEngineExtensions.cs (hunk 1 above)
Comment (markwallace-microsoft) [reply dotnet/src/Planners/Planners.Core/Handlebars/HandlebarsTemplateEngineExtensions.cs:361]: I will follow up on test coverage for the Handlebars planner

**Review by markwallace-microsoft — APPROVED** [review]

**Pull request merged**